libraries/botbuilder-ai/src/luis-client/luisClient.ts

```diff
@@ -0,0 +1,301 @@
+/**
+ * @module botbuilder-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/**
+ * NOTE: This class was auto generated by OpenAPI Generator (https://openapi-generator.tech)
+ * and was modify manually to make compliance with the current implementation of the library.
+ */
+
+import localVarRequest = require('request');
+import http = require('http');
+
+/* tslint:disable:no-unused-locals */
+import { LuisResult } from './model/luisResult';
+
+import { ObjectSerializer, Authentication, VoidAuth } from './model/models';
+import { ApiKeyAuth } from './model/models';
+
+let luisVersion = '/luis/v2.0';
+
+export interface PredictionResolveOptionalParams {
+    /**
+     * The timezone offset for the location of the request.
+     */
+    timezoneOffset?: number;
+    /**
+     * If true, return all intents instead of just the top scoring intent.
+     */
+    verbose?: boolean;
+    /**
+     * Use the staging endpoint slot.
+     */
+    staging?: boolean;
+    /**
+     * Enable spell checking.
+     */
+    spellCheck?: boolean;
+    /**
+     * The subscription key to use when enabling bing spell check
+     */
+    bingSpellCheckSubscriptionKey?: string;
+    /**
+     * Log query (default is true)
+     */
+    log?: boolean;
+
+    customHeaders: {headers: {[name: string]: string}};
+}
+
+export enum LuisApikeys {
+    apiKeyHeader,
+}
+
+export class LuisClient {
+    protected _basePath: string = '';
+    protected defaultHeaders: any = {};
```

Comment:
Is it ok that this is an `any`?

Reply:
Hi @Aliandi. The reason for the amount of `any` in this PR is because of these files were _auto-generated_ through _OpenAPI_, which tries to do its best. As the idea is to only generate them once and update the files with all the _things_ we need, we will review them and check which typed class/interface could we use instead of `any`.